diff --git a/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Liblist/Kid.pm b/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Liblist/Kid.pm
index fbdc79aea2..1e2f7f6c7d 100644
--- a/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Liblist/Kid.pm
+++ b/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Liblist/Kid.pm
@@ -84,6 +84,10 @@ sub _unix_os2_ext {
     }
     @libpath = grep -d, @libpath;
 
+    foreach my $path ( @libpath ) {
+        $ld_run_path_seen{$path}++;
+    }
+
     if ($^O eq 'darwin')  {
         # 'escape' Mach-O ld -framework and -F flags, so they aren't dropped later on
         $found++ if $potential_libs =~ s/(^|\s)(-(?:weak_|reexport_|lazy_)?framework)\s+(\S+)/$1-Wl,$2 -Wl,$3/g;
