--- icu/source/config/mh-linux	2025-03-14 00:01:23.000000000 +0530
+++ icu/source/config/mh-linux	2025-07-08 08:37:16.358169535 +0530
@@ -23,7 +23,7 @@
 LD_RPATH_PRE = -Wl,-rpath,
 
 ## These are the library specific LDFLAGS
-LDFLAGSICUDT=-nodefaultlibs -nostdlib
+LDFLAGSICUDT=
 
 ## Compiler switch to embed a library name
 # The initial tab in the next line is to prevent icu-config from reading it.
