From 80e469188094c1cb9206af0fbe74f469d1584468 Mon Sep 17 00:00:00 2001
From: Christian Rauch <Rauch.Christian@gmx.de>
Date: Thu, 3 Nov 2022 23:28:57 +0100
Subject: [PATCH] libdecor: print an error message when a plugin has a
 conflicting symbol

---
 src/libdecor.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/libdecor.c b/src/libdecor.c
index 7d48d14..a9c1106 100644
--- a/src/libdecor.c
+++ b/src/libdecor.c
@@ -1397,8 +1397,11 @@ check_symbol_conflicts(const struct libdecor_plugin_description *plugin_descript
 	while (*symbol) {
 		dlerror();
 		dlsym (RTLD_DEFAULT, *symbol);
-		if (!dlerror())
+		if (!dlerror()) {
+			fprintf(stderr, "Plugin \"%s\" uses conflicting symbol \"%s\".\n",
+					plugin_description->description, *symbol);
 			return false;
+		}
 
 		symbol++;
 	}
-- 
GitLab

