diff --git a/modules.d/99systemd-sysusers/module-setup.sh b/modules.d/99systemd-sysusers/module-setup.sh
index c1ef40cd..1618f80b 100755
--- a/modules.d/99systemd-sysusers/module-setup.sh
+++ b/modules.d/99systemd-sysusers/module-setup.sh
@@ -23,4 +23,5 @@ install() {
         set -o pipefail
         systemd-sysusers --root="$initdir" 2>&1 >&3 | grep -v "^Creating " >&2
     } 3>&1
+    chmod 0400 "${initdir}/etc/gshadow" "${initdir}/etc/shadow" || true
 }
