diff --git a/modules.d/60systemd-sysusers/module-setup.sh b/modules.d/60systemd-sysusers/module-setup.sh
index 05680553..cd459a3b 100755
--- a/modules.d/60systemd-sysusers/module-setup.sh
+++ b/modules.d/60systemd-sysusers/module-setup.sh
@@ -16,4 +16,5 @@ install() {
     inst_sysusers basic.conf
 
     systemd-sysusers --root="$initdir"
+    chmod 0400 "${initdir}/etc/gshadow" "${initdir}/etc/shadow" || true
 }
