From 7e070855f6446d372cedde2e05332b3fac155ae7 Mon Sep 17 00:00:00 2001
From: Seppo Yli-Olli <seppo.yliolli@gmail.com>
Date: Wed, 9 Oct 2024 22:40:39 +0300
Subject: [PATCH] Fix log default not to be int

---
 common/buildboxcommon_logging.cpp | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/common/buildboxcommon_logging.cpp b/common/buildboxcommon_logging.cpp
index 67c575f70..3681c9ff2 100644
--- a/common/buildboxcommon_logging.cpp
+++ b/common/buildboxcommon_logging.cpp
@@ -128,17 +128,14 @@ void Logger::initialize(const char *programName)
 
 void Logger::setLogLevel(LogLevel logLevel)
 {
-    google::LogSeverity glogMinSeverity = 0;
+    /* By default show all messages */
+    google::LogSeverity glogMinSeverity = google::GLOG_INFO;
     if (logLevel == buildboxcommon::LogLevel::WARNING) {
         glogMinSeverity = google::GLOG_WARNING;
     }
     else if (logLevel == buildboxcommon::LogLevel::ERROR) {
         glogMinSeverity = google::GLOG_ERROR;
     }
-    else {
-        /* By default show all messages */
-        glogMinSeverity = google::GLOG_INFO;
-    }
 
     FLAGS_minloglevel = glogMinSeverity;
 
-- 
GitLab

