diff --git a/src/pipewire/stream.c b/src/pipewire/stream.c
index 9ba438b6a..50d0e23a1 100644
--- a/src/pipewire/stream.c
+++ b/src/pipewire/stream.c
@@ -612,8 +612,10 @@ static int impl_send_command(void *object, const struct spa_command *command)
 		if (stream->state == PW_STREAM_STATE_PAUSED) {
 			pw_log_debug("%p: start %d", stream, impl->direction);
 
-			if (impl->direction == SPA_DIRECTION_INPUT)
-				impl->io->status = SPA_STATUS_NEED_DATA;
+			if (impl->direction == SPA_DIRECTION_INPUT) {
+				if (impl->io != NULL)
+					impl->io->status = SPA_STATUS_NEED_DATA;
+			}
 			else if (!impl->process_rt && !impl->driving)
 				call_process(impl);
 
@@ -992,6 +994,9 @@ static int impl_node_process_input(void *object)
 	struct spa_io_buffers *io = impl->io;
 	struct buffer *b;
 
+	if (io == NULL)
+		return -EIO;
+
 	pw_log_trace_fp("%p: process in status:%d id:%d ticks:%"PRIu64" delay:%"PRIi64,
 			stream, io->status, io->buffer_id, impl->time.ticks, impl->time.delay);
 
@@ -1388,7 +1393,8 @@ static void context_drained(void *data, struct pw_impl_node *node)
 		return;
 	if (impl->draining && impl->drained) {
 		impl->draining = false;
-		impl->io->status = SPA_STATUS_NEED_DATA;
+		if (impl->io != NULL)
+			impl->io->status = SPA_STATUS_NEED_DATA;
 		call_drained(impl);
 	}
 }
