diff --git a/packaging/pep517_backend/_backend.py b/packaging/pep517_backend/_backend.py
index a163f9c..3dbb4f6 100644
--- a/packaging/pep517_backend/_backend.py
+++ b/packaging/pep517_backend/_backend.py
@@ -376,7 +376,7 @@ def get_requires_for_build_wheel(
     if is_pure_python_build:
         c_ext_build_deps = []
     else:
-        c_ext_build_deps = ['Cython ~= 3.1.0']
+        c_ext_build_deps = ['Cython >= 3.0.12']
 
     return _setuptools_get_requires_for_build_wheel(
         config_settings=config_settings,
diff --git a/requirements/cython.txt b/requirements/cython.txt
index c3ca2f2..3d6bf1b 100644
--- a/requirements/cython.txt
+++ b/requirements/cython.txt
@@ -1 +1 @@
-cython==3.1.2
+cython>=3.0.12
