diff --git a/packaging/pep517_backend/_backend.py b/packaging/pep517_backend/_backend.py
index f4dcb33..cf16421 100644
--- a/packaging/pep517_backend/_backend.py
+++ b/packaging/pep517_backend/_backend.py
@@ -413,7 +413,7 @@ def get_requires_for_build_wheel(
     if is_pure_python_build:
         c_ext_build_deps = []
     else:
-        c_ext_build_deps = ['Cython >= 3.2.0']
+        c_ext_build_deps = ['Cython >= 3.0.12']
 
     return _setuptools_get_requires_for_build_wheel(
         config_settings=config_settings,
