diff --git a/configure.ac b/configure.ac
index b583d16ab..e27644c53 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1996,7 +1996,6 @@ tests/openpgp/Makefile
 tests/migrations/Makefile
 tests/gpgsm/Makefile
 tests/gpgme/Makefile
-tests/pkits/Makefile
 g10/gpg.w32-manifest
 g10/gpgv.w32-manifest
 sm/gpgsm.w32-manifest
diff --git a/tests/Makefile.am b/tests/Makefile.am
index b5749f9ef..80f90c085 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -18,7 +18,7 @@
 
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = gpgscm openpgp migrations gpgsm gpgme pkits .
+SUBDIRS = gpgscm openpgp migrations gpgsm gpgme .
 
 GPGSM = ../sm/gpgsm
 
