diff --git a/conf.d/65-nonlatin.conf b/conf.d/65-nonlatin.conf
index 3e5d1c7d..e18f9b0f 100644
--- a/conf.d/65-nonlatin.conf
+++ b/conf.d/65-nonlatin.conf
@@ -68,7 +68,7 @@
 			<family>Lohit Malayalam</family>
  			<family>Lohit Kannada</family>
  			<family>Lohit Telugu</family>
- 			<family>Lohit Oriya</family>
+ 			<family>Lohit Odia</family>
  			<family>LKLUG</family>
 		</prefer>
 	</alias>
@@ -149,7 +149,7 @@
  			<family>Lohit Malayalam</family>
  			<family>Lohit Kannada</family>
  			<family>Lohit Telugu</family>
- 			<family>Lohit Oriya</family>
+ 			<family>Lohit Odia</family>
  			<family>LKLUG</family>
 		</prefer>
 	</alias>
@@ -200,7 +200,7 @@
 			<family>Lohit Malayalam</family>
 			<family>Lohit Kannada</family>
 			<family>Lohit Telugu</family>
-			<family>Lohit Oriya</family>
+			<family>Lohit Odia</family>
 			<family>LKLUG</family>
 		</prefer>
 	</alias>
