diff --git a/x86/insns-iflags.ph b/x86/insns-iflags.ph
index ff5b1997..e122c24f 100644
--- a/x86/insns-iflags.ph
+++ b/x86/insns-iflags.ph
@@ -248,7 +248,7 @@ sub merge_flags($;$) {
 
     clean_flags(\%flags);
 
-    my @flagslist = sort { $flag_byname{$a} <=> $flag_byname{$b} }
+    my @flagslist = sort { $flag_byname{$a}->[0] <=> $flag_byname{$b}->[0] }
 	grep { !/^(\s*|\!.*)$/ } keys(%$flags);
 
     if ($human) {
