diff -up aom-3.12.0/build/cmake/aom_optimization.cmake.nasm3 aom-3.12.0/build/cmake/aom_optimization.cmake
--- aom-3.12.0/build/cmake/aom_optimization.cmake.nasm3 2025-02-11 02:37:18.000000000 +0100
+++ aom-3.12.0/build/cmake/aom_optimization.cmake       2025-10-14 10:19:02.551479564 +0200
@@ -212,7 +212,16 @@ endfunction()
 # Currently checks only for presence of required object formats and support for
 # the -Ox argument (multipass optimization).
 function(test_nasm)
-  execute_process(COMMAND ${CMAKE_ASM_NASM_COMPILER} -hf
+  execute_process(COMMAND ${CMAKE_ASM_NASM_COMPILER} -v
+                  OUTPUT_VARIABLE nasm_version_output)
+  string(REGEX MATCH "[0-9]+\.[0-9]+\.[0-9]+" nasm_version ${nasm_version_output})
+
+  if(nasm_version VERSION_GREATER_EQUAL 3.0)
+    set(nasm_help_option "-h all")
+  else()
+    set(nasm_help_option -hf)
+  endif()
+  execute_process(COMMAND ${CMAKE_ASM_NASM_COMPILER} ${nasm_help_option}
                   OUTPUT_VARIABLE nasm_helptext)

   if(NOT "${nasm_helptext}" MATCHES "-Ox")
